### ISSUE TITLE
test_fiiter fails on 3.8 but passes on < 3.8

### ISSUE THREAD

**Issue by sjsrey** [pysal/libpysal#249]
```
    def test_filter(self):

        grid = lat2W(2,2)

        alist = grid.to_adjlist(remove_symmetric=True)

        assert len(alist) == 4

        with self.assertRaises(AssertionError):

            badgrid = weights.W.from_adjlist(alist)

            np.testing.assert_allclose(badgrid.sparse.toarray(),

                                       grid.sparse.toarray())

>       assert set(alist.focal.unique().tolist()) == set(list(range(4)))

E       AssertionError: assert {0, 1, 3} == {0, 1, 2, 3}

E         Extra items in the right set:

E         2

E         Full diff:

E         - {0, 1, 3}

E         + {0, 1, 2, 3}

E         ?        +++

```

[source](https://travis-ci.org/sjsrey/libpysal/jobs/651597585#L2511)

**Comment by ljwolf** [comment]
I think this is solved by #244?

**Comment by sjsrey** [comment]
Thanks for reminding me. Will merge that and reassign this to me.

**Comment by sjsrey** [comment]
Fixed with #251
